test/SimpleSlippageRouter.t.sol

```diff
@@ -0,0 +1,459 @@
+
+        // Setup pricing
+        PriceRouter.ChainlinkDerivativeStorage memory stor;
+
+        PriceRouter.AssetSettings memory settings;
+
+        uint256 price = uint256(mockUsdcUsd.latestAnswer());
+        settings = PriceRouter.AssetSettings(CHAINLINK_DERIVATIVE, address(mockUsdcUsd));
+        priceRouter.addAsset(USDC, settings, abi.encode(stor), price);
+
+        // Setup exchange rates:
+        // USDC Simulated Price: $1
+        // mockUsdcUsd.setMockAnswer(1e6);
+
+        // Add adaptors and ERC20 positions to the registry.
+        registry.trustPosition(usdcPosition, address(erc20Adaptor), abi.encode(USDC));
+
+        // Create Dummy Cellars.
+        string memory cellarName = "Dummy Cellar V0.0";
+        uint256 initialDeposit = 1e6;
+        uint64 platformCut = 0.75e18;
+
+        cellarName = "Cellar V0.0";
+        initialDeposit = 1e6;
+        platformCut = 0.75e18;
+        cellar = _createCellar(cellarName, USDC, usdcPosition, abi.encode(0), initialDeposit, platformCut);
+
+        vm.label(address(cellar), "cellar");
+
+        // Approve cellar to spend all assets.
+        USDC.approve(address(cellar), type(uint256).max);
+        USDC.approve(address(simpleSlippageRouter), type(uint256).max);
+
+        initialAssets = cellar.totalAssets();
+        initialShares = cellar.totalSupply();
+    }
+
+    // ========================================= INITIALIZATION TEST =========================================
+
+    // simpleSlippageRouter should not have any assets, any approvals, etc.
+    function testInitialization() external {}
+
+    // ========================================= HAPPY PATH TEST =========================================
+
+    // test depositing using SimpleSlippageRouter
+    // deposit() using SSR, deposit again using SSR. See that appropriate amount of funds were deposited.
+    function testDeposit(uint256 assets) external {
+        assets = bound(assets, 1e6, 100_000e6);
+
+        // deal USDC assets to test contract
+        deal(address(USDC), address(this), assets);
+        deposit1 = assets / 2;
+        minShares1 = deposit1;
+        deadline1 = uint64(block.timestamp + 1 days);
+
+        console.log("BEFORE: Total Cellar Shares: %s", cellar.totalSupply());
+
+        console.log("BEFORE: Cellar Shares Belonging to Test Contract: %s", cellar.balanceOf(address(this)));
+
+        // deposit half using the SSR
+        simpleSlippageRouter.deposit(cellar, deposit1, minShares1, deadline1);
+        console.log("AFTER: Cellar Shares Belonging to Test Contract: %s", cellar.balanceOf(address(this)));
+        console.log("AFTER: Total Cellar Shares: %s", cellar.totalSupply());
+
+        shareBalance1 = cellar.balanceOf(address(this));
+
+        assertEq(shareBalance1, minShares1); // shares and assets are 1:1 right now because it's just USDC in a holding position - TODO: initialShares belongs to  this test contract since it created the cellar no?
```

Comment:
Thanks for the confirmation, okay so that's what I though too, but when I run the assert with the sum of `minShares1 + initialShares` it fails for some reason.